src/mxarray.jl

```diff
@@ -655,10 +655,10 @@ function jstring(mx::MxArray)
         throw(ArgumentError("jstring only applies to strings (i.e. char vectors)."))
     end
     len = ncols(mx) + 2
-    tmp = Array(UInt8, len)
+    tmp = Array(UInt8, len-2)
     ccall(_mx_get_string, Cint, (Ptr{Void}, Ptr{UInt8}, mwSize), 
         mx.ptr, tmp, len)
-    bytestring(pointer(tmp))
+    String(tmp)
```

Comment:
Assuming it writes a NUL terminated string, it's fine to write pass the end of a directly allocated byte array. Writing non-NUL or writing more than 1 byte is not fine though.